Documentation for creation of custom toolkit (#2715)
VirtusLab/scala-cli: website/docs/guides/advanced/custom-toolkit.md

title: Custom Toolkit
sidebar_position: 44
---

> Similar to the Scala Toolkit and Typelevel toolkit, it is possible to create our own custom toolkit. 

Reviewer: ```suggestion
Similar to the Scala Toolkit and Typelevel toolkit, it is possible to create your own, custom toolkit. 
```

Having a custom toolkit with common libraries can speed up the development using scala-cli. 

Let's look at how we can create a new toolkit. 

For example, to create a LiHaoyi ecosystem toolkit, we can name the file as `LiHaoyiToolkit.scala` and add the required libraries as dependency directives:

> ```

Reviewer: ```suggestion
```scala
```

Similarly, define the scalajs version of toolkit in `LiHaoyiToolkit.js.scala` file. Notice the `js.scala` extension. It should also have `publish.name` as `toolkit`. 

If testkit is supported, it can also be added as another file, `LiHaoyiToolkitTest.scala` with `publish.name` as `toolkit-test`:
```
//> using scala 2.13, 3
//> using publish.name toolkit-test
//> using dep com.lihaoyi::utest::0.8.2
```

Additionally, more configurations needed for publishing the toolkit can be kept in a conf file, for example, `publish-conf.scala`:
```
//> using publish.organization com.yadavan88
//> using publish.version 0.1.0
//> using publish.url https://github.com/yadavan88/lihaoyi-toolkit
//> using publish.license Apache-2.0
//> using publish.repository central
//> using publish.developer "yadavan88|Yadu Krishnan|https://github.com/yadavan88"
//> using repository sonatype:public
```

The toolkit can be published locally using the command:
```
scala-cli --power publish local --cross LiHaoyiToolkit.scala publish-conf.scala
```

> In the similar way, it is also possible to publish to a central repository. Refer to the [GitHub Action workflow](https://github.com/scala/toolkit/blob/main/.github/workflows/deploy.yaml) for more details.

Reviewer: ```suggestion
Similarly, it is also possible to publish to a central repository. Refer to the [GitHub Action workflow](https://github.com/scala/toolkit/blob/main/.github/workflows/deploy.yaml) for more details.
```
